config/src/main/java/com/quorum/tessera/config/constraints/ServerConfigsValidator.java

```diff
@@ -1,45 +1,57 @@
 package com.quorum.tessera.config.constraints;
 
 import com.quorum.tessera.config.AppType;
+import com.quorum.tessera.config.Config;
 import com.quorum.tessera.config.ServerConfig;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.validation.ConstraintValidator;
 import javax.validation.ConstraintValidatorContext;
-import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
 
-public class ServerConfigsValidator implements ConstraintValidator<ValidServerConfigs, List<ServerConfig>> {
+public class ServerConfigsValidator implements ConstraintValidator<ValidServerConfigs, Config> {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ServerConfigsValidator.class);
 
     @Override
-    public boolean isValid(List<ServerConfig> serverConfigs, ConstraintValidatorContext constraintContext) {
-        if(serverConfigs == null) {
+    public boolean isValid(Config config, ConstraintValidatorContext constraintContext) {
+        if (config == null || config.getServerConfigs() == null || config.getServerConfigs().isEmpty()) {
             return true;
         }
 
-        final Map<AppType,Integer> counts = serverConfigs.stream()
-            .collect(Collectors.toMap(ServerConfig::getApp,v -> 1,(l,r) -> l + 1));
+        final Map<AppType, Integer> counts =
+                config.getServerConfigs().stream()
+                        .collect(Collectors.toMap(ServerConfig::getApp, v -> 1, (l, r) -> l + 1));
 
-        final int p2PEnabledConfigsCount = counts.getOrDefault(AppType.P2P,0);
-        final int q2TEnabledConfigsCount = counts.getOrDefault(AppType.Q2T,0);
+        final int p2PEnabledConfigsCount = counts.getOrDefault(AppType.P2P, 0);
+        final int q2TEnabledConfigsCount = counts.getOrDefault(AppType.Q2T, 0);
 
         if (p2PEnabledConfigsCount != 1) {
             LOGGER.debug("Only one P2P server must be configured and enabled.");
             constraintContext.disableDefaultConstraintViolation();
-            constraintContext.buildConstraintViolationWithTemplate("Only one P2P server must be configured and enabled.")
-                .addConstraintViolation();
+            constraintContext
+                    .buildConstraintViolationWithTemplate("Only one P2P server must be configured and enabled.")
```

Comment:
Should we change this message to from `Only` to `Exactly`? It reads a bit funny if no P2P servers are defined, since it implies multiple have been.

Reply:
`and enabled` I think can be removed since we removed the ability for a server to be in a disabled state.

Reply:
Same for the log msg on line 32